the-alexa-skill/typescript/lib/the-alexa-skill-stack.ts

```diff
@@ -0,0 +1,96 @@
+import * as cdk from '@aws-cdk/core';
+import lambda = require('@aws-cdk/aws-lambda');
+import dynamodb = require('@aws-cdk/aws-dynamodb');
+import * as alexaAsk from '@aws-cdk/alexa-ask';
+import { ServicePrincipal, Role, PolicyStatement, CompositePrincipal } from '@aws-cdk/aws-iam';
+
+
+export interface AssetProps extends cdk.StackProps {
+  readonly assetBucketARN: string;
+  readonly assetBucketName: string;
+  readonly assetObjectKey: string;
+}
+
+export class TheAlexaSkillStack extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props: AssetProps) {
+    super(scope, id, props);
+
+    //role to access bucket
+    const role = new Role(this, 'Role', {
+      assumedBy:new CompositePrincipal(
+        new ServicePrincipal('alexa-appkit.amazon.com'),
+        new ServicePrincipal('cloudformation.amazonaws.com')
+      )
+    });
+
+
+    // Allow the skill resource to access the zipped skill package
+    role.addToPolicy(new PolicyStatement({
+      actions: ['S3:GetObject'],
+      resources: [props.assetBucketARN+'/'+props.assetObjectKey]
+    }));
+
+    // DynamoDB Table
+     const usersTable = new dynamodb.Table(this, 'Users', {
+      partitionKey: { name: 'userId', type: dynamodb.AttributeType.STRING },
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST
+    });
+
+    // Lambda function for Alexa fulfillment
+    const alexaLambda = new lambda.Function(this, 'AlexaLambdaHandler', {
+      runtime: lambda.Runtime.NODEJS_12_X,
+      code: lambda.Code.asset('lambda-fns'),
+      handler: 'lambda.handler',
+      environment: {
+        USERS_TABLE: usersTable.tableName
+      }
+    });
+
+    // grant the lambda role read/write permissions to our table
+    usersTable.grantReadWriteData(alexaLambda);
+
+    // create the skill /* TODO identify something unique that can trigger this to look for new contents at the S3 Key */
+    const skill = new alexaAsk.CfnSkill(this, 'the-alexa-skill', {
```

Comment:
I haven't deployed this yet to check but does this mean that if you wanted to change anything from that skill folder which is uploaded to S3 that currently you would need to delete the asset stack and redeploy both stacks to get it to pick it up?

Reply:
That is how it originally was before I figured out how to properly upload the skill assets. Now if you update anything in the skill folder, the asset stack uploads a new zip file and the skill deploy picks that up. I will fix the comment.